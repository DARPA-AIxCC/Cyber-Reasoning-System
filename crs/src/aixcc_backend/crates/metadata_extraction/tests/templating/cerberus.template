[
  {%- for harness_id, harness in harnesses %}
  {
    "id": {{loop.index}},
    "name": "{{cp_name}}",
    "subject": "{{cp_sanitized}}",
    "bug_id": "{{harness_id}}_{{loop.index}}",
    "cp-abspath": "",
    "bug_type": "{{bug_type}}",
    "sanitizers": [
      {% for sanitizer_id, sanitizer_name in sanitizers -%}
      {
        "id": "{{sanitizer_id}}",
        "name": "{{sanitizer_name}}"
      }
      {%- if loop.last == false%},{% endif -%}
      {%- endfor %}
    ],
    "base_image": "{{docker_image}}",
    "cp_sources": [
      {% for source_entry in cp_sources -%}
      {
        "name": "{{source_entry.path}}",
        "address": "{{source_entry.address}}",
        "ref": "{{source_entry.source_ref}}"
      }{% if loop.last == false%},{% endif %}
      {%- endfor %}
    ],
    "harnesses": [
      {
        "name": "{{harness.name}}",
        "source": "{{harness.source}}",
        "binary": "{{harness.binary}}"
      }
    ],
    "analysis_output": [],
    "bug_reports": [],
    "binary_path": "{{harness.binary}}",
    "test_timeout": {{test_timeout}},
    "language": "{{language | lower_case }}",
    "build_script": "{{build_script}}",
    "config_script": "{{config_script}}",
    "test_script": "{{test_script}}",
    "instrument_script": "{{instrument_script}}",
    "trace_script": "{{trace_script}}",
    "validate_script": "{{validate_script}}",
    "build_script_internal": "{{build_script_internal}}",
    "run_script_internal": "{{run_script_internal}}",
    "test_script_internal": "{{test_script_internal}}",
    "output_dir_abspath": "",
    "pvt_test_script": "",
    "source_file": "",
    "line_numbers": [],
    "localization": [],
    "checkout_commit": "{{cp_sources[0].source_ref}}",
    "build_command": "",
    "compile_programs": [],
    "failing_test_identifiers": [],
    "passing_test_identifiers": [],
    "count_neg": 0,
    "count_pos": 0
  }
  {%- if loop.last == false%},{% endif -%}
  {% endfor %}
]