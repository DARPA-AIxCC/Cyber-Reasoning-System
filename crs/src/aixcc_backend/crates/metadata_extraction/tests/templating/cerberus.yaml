cp_name: "linux kernel"

# The language property denotes the *primary* language used; vulnerabilities will
# exist solely within code of the CP's primary langauge.

language: "c"

# The cp_address property points the reader to the full challenge project repository
# this exists so that the project.yaml file may be passed around standalone.

cp_address: "https://github.com/aixcc-public/challenge-001-exemplar.git"

# The cp_source_address property denotes the internal source repository addresses to use

cp_source_address:
  linux-kernel: "https://github.com/aixcc-public/challenge-001-exemplar-source.git"

# The docker_img_address property denotes the address to pull the pre-built docker
# image that the CP's Dockerfile bases from.

docker_img_address: "n/a"

# The sanitizers property denotes which sanitizers are available for scoring for
# this particular challenge project; when specifying a santizer id to the scoring
# server, use the *key* values here, e.g. use "id_1", not "sanitizer-1: subtype-1".
#
# Note, these key-value mappings are unique for each challenge project, sanitizer
# id "id_1" will likely map to different values for each challenge project.

sanitizers:
  id_1: "KASAN: slab-out-of-bounds"

# The harnesses property denotes which aixcc harnesses are available to use for
# triggering sanitizers for this challenge project; when specifying a harness id
# to the scoring server, use the *key* value here, e.g. "test_harness_id_1", not
# the values under 'source' or 'binary'.
#
# The source and binary properties point the reader to the appropriate directory
# paths for the harness source code and executable binary (if exists).

harnesses:
  linux_test_harness:
    source: "src/test_harnesses/linux_test_harness.c"
    binary: "out/linux_test_harness"
  linux_test_harness2:
    source: "src/test_harnesses/linux_test_harness.c2"
    binary: "out/linux_test_harness2"