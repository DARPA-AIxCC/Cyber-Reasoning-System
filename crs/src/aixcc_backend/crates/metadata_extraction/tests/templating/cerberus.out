[
  {
    "id": 1,
    "name": "linux kernel",
    "subject": "**cp_sanitized**",
    "bug_id": "linux_test_harness_1",
    "cp-abspath": "",
    "bug_type": "test-failure",
    "sanitizers": [
      {
        "id": "id_1",
        "name": "KASAN: slab-out-of-bounds"
      }
    ],
    "base_image": "",
    "cp_sources": [
      {
        "name": "linux-kernel",
        "address": "https://github.com/aixcc-public/challenge-001-exemplar-source.git",
        "ref": "main"
      }
    ],
    "harnesses": [
      {
        "name": "linux_test_harness",
        "source": "src/test_harnesses/linux_test_harness.c",
        "binary": "out/linux_test_harness"
      }
    ],
    "analysis_output": [],
    "bug_reports": [],
    "binary_path": "out/linux_test_harness",
    "test_timeout": 10,
    "language": "c",
    "build_script": "build.sh",
    "config_script": "config.sh",
    "test_script": "test.sh",
    "instrument_script": "**instrument_script**",
    "trace_script": "**trace_script**",
    "validate_script": "**validate_script**",
    "build_script_internal": "**build_script_internal**",
    "run_script_internal": "**run_script_internal**",
    "test_script_internal": "**test_script_internal**",
    "output_dir_abspath": "",
    "pvt_test_script": "",
    "source_file": "",
    "line_numbers": [],
    "localization": [],
    "checkout_commit": "main",
    "build_command": "",
    "compile_programs": [],
    "failing_test_identifiers": [],
    "passing_test_identifiers": [],
    "count_neg": 0,
    "count_pos": 0
  },
  {
    "id": 2,
    "name": "linux kernel",
    "subject": "**cp_sanitized**",
    "bug_id": "linux_test_harness2_2",
    "cp-abspath": "",
    "bug_type": "test-failure",
    "sanitizers": [
      {
        "id": "id_1",
        "name": "KASAN: slab-out-of-bounds"
      }
    ],
    "base_image": "",
    "cp_sources": [
      {
        "name": "linux-kernel",
        "address": "https://github.com/aixcc-public/challenge-001-exemplar-source.git",
        "ref": "main"
      }
    ],
    "harnesses": [
      {
        "name": "linux_test_harness2",
        "source": "src/test_harnesses/linux_test_harness.c2",
        "binary": "out/linux_test_harness2"
      }
    ],
    "analysis_output": [],
    "bug_reports": [],
    "binary_path": "out/linux_test_harness2",
    "test_timeout": 10,
    "language": "c",
    "build_script": "build.sh",
    "config_script": "config.sh",
    "test_script": "test.sh",
    "instrument_script": "**instrument_script**",
    "trace_script": "**trace_script**",
    "validate_script": "**validate_script**",
    "build_script_internal": "**build_script_internal**",
    "run_script_internal": "**run_script_internal**",
    "test_script_internal": "**test_script_internal**",
    "output_dir_abspath": "",
    "pvt_test_script": "",
    "source_file": "",
    "line_numbers": [],
    "localization": [],
    "checkout_commit": "main",
    "build_command": "",
    "compile_programs": [],
    "failing_test_identifiers": [],
    "passing_test_identifiers": [],
    "count_neg": 0,
    "count_pos": 0
  }
]